tests/src/webhooks/handleInvoiceCompleted.test.ts

```diff
@@ -375,4 +499,99 @@ describe('Process when an invoice payment is completed', () => {
+        customer: mockedCustomer.id,
+        lines: {
+          data: [
+            {
+              price: {
+                id: `price_12333`,
+                object: 'price',
+                active: true,
+                billing_scheme: 'per_unit',
+                created: 102389234,
+                currency: 'usd',
+                custom_unit_amount: null,
+                livemode: false,
+                lookup_key: null,
+                metadata: {
+                  maxSpaceBytes: `1837284738`,
+                  type: UserType.Individual,
+                },
+                nickname: null,
+                product: {
+                  id: `prod_12333`,
+                  type: 'service',
+                  object: 'product',
+                  active: true,
+                  created: 1678833149,
+                  default_price: null,
+                  description: null,
+                  images: [],
+                  marketing_features: [],
+                  livemode: false,
+                  metadata: {
+                    type: UserType.Individual,
+                  },
+                  name: 'Gold Plan',
+                  package_dimensions: null,
+                  shippable: null,
+                  statement_descriptor: null,
+                  tax_code: null,
+                  unit_label: null,
+                  updated: 1678833149,
+                  url: null,
+                },
+                recurring: {
+                  aggregate_usage: null,
+                  interval: 'month',
+                  interval_count: 1,
+                  trial_period_days: null,
+                  usage_type: 'licensed',
+                },
+                tax_behavior: 'unspecified',
+                tiers_mode: null,
+                transform_quantity: null,
+                type: 'recurring',
+                unit_amount: 1000,
+                unit_amount_decimal: '1000',
+              },
+              discounts: [
+                {
+                  id: 'coupon_id',
+                  coupon: 'COUPON' as any,
+                },
+              ],
+            },
+          ],
+        },
+      } as any);
+      jest.spyOn(paymentService, 'getCustomer').mockResolvedValue(mockedCustomer as any);
+      jest.spyOn(paymentService, 'getInvoiceLineItems').mockResolvedValue(mockedInvoice.lines as any);
+      jest.spyOn(usersService, 'findUserByCustomerID').mockResolvedValue(mockedUser);
+      jest.spyOn(usersRepository, 'updateUser').mockImplementation();
+      jest.spyOn(usersService, 'findUserByUuid').mockResolvedValue(mockedUser);
+      const storedCouponSpy = jest.spyOn(usersService, 'storeCouponUsedByUser').mockResolvedValue();
+
+      await handleInvoiceCompleted(
+        mockedInvoice,
+        usersService,
+        paymentService,
+        getLogger(),
+        cacheService,
+        config,
+        objectStorageService,
+      );
+
+      expect(storedCouponSpy).toHaveBeenCalledTimes(1);
```

Comment:
I am approving it but please do check **always** the parameters of the function you are testing are correct. Assuming that because the execution reached this point is enough is making half of the work to ensure a strict test, if parameters are wrong, a lot of bugs have room to get in here even having this test in-between so the test become useless